src/plugins/gstreamer/camerabin/camerabinimageprocessing.cpp

```diff
@@ -313,7 +313,10 @@ bool CameraBinImageProcessing::isParameterValueSupported(QCameraImageProcessingC
 
                 for (int i = 0; i < effect_count; i++) {
                     GVariant *mode = g_variant_get_child_value(supported_effects, i);
-                    supportedEffects << m_filterMap.keys().at(m_filterMap.values().indexOf((GstPhotographyColorToneMode)g_variant_get_int32(mode)));
+                    int index = m_filterMap.values().indexOf((GstPhotographyColorToneMode)g_variant_get_int32(mode));
```

Comment:
Not knowing much about what's going on, value => key lookup looks like a strange way to use a hash table.

Reply:
The build a list to see if contains a value seems a bit strange too. Would seem more efficient to take the expected value from the map and then just see if the array contains that value.

Furthermore, all these variants should probably be disposed of.

Reply:
I reworked both color filter and white balance loops which both used temporary lists.